capsules/src/nonvolatile_storage_driver.rs

```diff
@@ -242,14 +243,14 @@ impl<'a> NonvolatileStorage<'a> {
 
                                 // Need to copy bytes if this is a write!
                                 if command == NonvolatileCommand::UserspaceWrite {
-                                    app.buffer_write.as_mut().map(|app_buffer| {
+                                    app.buffer_write.map_or((), |app_buffer| {
```

Comment:
flagging `map_or` with ignored return value for later evaluation / possible fix

Reply:
Right, like in this case we want to return how much was actually written. (line 251).  And note that line 247 means that you could actually write nothing, yet never tell anyone this happened.